swagger/swagger-invocation/invocation-core/src/main/java/io/servicecomb/swagger/engine/SwaggerEnvironment.java

```diff
@@ -45,14 +47,20 @@
   @Inject
   private ProducerArgumentsMapperFactory producerArgumentsFactory;
 
-  @Inject
-  private ProducerResponseMapperFactory producerResponseMapperFactory;
+  private ResponseMapperFactorys<ProducerResponseMapper> producerResponseMapperFactorys =
```

Comment:
why do we make this kind of change?

Reply:
1.old mechanism only support one level mapper, like:
`ResponseEntity<String> -> String -> ResponseEntity<String>`

but after we support CompletableFuture, we need multiple level mapper, like:
`CompletableFuture<ResponseEntity<String>> -> String -> CompletableFuture<ResponseEntity<String>>`

2.old mechanism is complex and not flexible, hard to extend